content/concept-history-sharding.md

```diff
@@ -0,0 +1,53 @@
+
+## Comparing the Node Store and Shard Store
+
+Like the `node store`, the `shard store` derives from the base class `Database`. The common interface facilitates replacing the `node store` when storing or fetching from various places in the project. For example, the inbound ledger process uses synchronization filters to save data received from peers to either store. Specifying the database object for the filter allows `rippled` to decide which store receives the data. Similarly, a `Family` tells `SHAMaps` which database to use when fetching or storing. The derived stores themselves rely on these commonalities when copying ledgers from one store to another or validating their state and transaction trees.
+
+## XRP Ledger Network Data Integrity
+
+The history of all ledgers is shared by servers agreeing to keep particular ranges of historical ledgers. This makes it possible for servers to confirm that they have all the data they agreed to maintain, and produce proof trees or ledger deltas. Sharding also makes it possible to challenge servers to demonstrate they hold the data they claim to have, and verify their proofs.
+
+## Shard Configuration
+Rippled can be configured to begin storing ledger history in shards by adding the `shard_db` section to the `rippled.cfg` file.
+
+### Shard Configuration Example
+The following snippet from an example `rippled.cfg` file shows the configuration fields for adding sharding to a `rippled` server:
+
+```
+# Begin Shard support
+[shard_db]
+type=NuDB
+path=/Users/{username}/.rippled/db/shards/nudb
+max_size_gb=50
+# End shard support
+```
+
+**Tip:** Ripple recommends using NuDB for sharding (`type=NuDB`). NuDB uses fewer file handles per shard than RocksDB, while RocksDB uses memory that scales with the size of data its storing, which, with many shards may require excessive memory overhead.
+
+For more information, reference the `[shard_db]` example in the [rippled.cfg configuration example](https://github.com/ripple/rippled/blob/master/doc/rippled-example.cfg).
+
+### Sizing the Shard Store
+Determining a suitable value for the shard store will involve carefully considering different aspects of the process. Although redundant, it is entirely possible to hold full history in the node store and the shard store. Servers can shrink the range of ledgers they hold by dropping the shard database and then re-acquiring ledgers through the acquisition process.
+
+The following concerns should each be considered in determining shard store sizing:
+
+- Each shard of the node store is 2^14 ledgers (at time of publishing, Q1 2018), which may not be optimal.
+- An effective configuration might limit the node store only to recent history.
+- The node store history size should at minimum be twice the ledgers per shard, due to the fact that the current shard may be chosen to be stored and it would be wasteful to reacquire that data.
```

Comment:
Passive voice. Also, when you say, "the current shard", do you mean the most recent ledger data?